lib/commands/setup.js

```diff
@@ -147,7 +147,15 @@ class SetupCommand extends Command {
             }
 
             return this.ui.listr(tasks, {setup: true}).then(() => {
-                if (!argv.prompt || argv.start) {
+                // If we are not allowed to prompt, set the default value, which should be true
+                if (!argv.prompt) {
+                    argv.start = true;
+                }
+
+                // If argv.start has a value, this means either --start or --no-start were explicitly provided
+                // (or --no-prompt was provided, and we already defaulted to true)
+                // In this case, we don't prompt, we use the value of argv.start
+                if (argv.start) {
```

Comment:
if `--no-start` is explicitly provided, `argv.start` will still be false, and this if statement wouldn't execute.